telluric/georaster.py

```diff
@@ -206,27 +207,24 @@ def _merge(one, other, merge_strategy=MergeStrategy.UNION):
         other_remaining_bands = [band for band in other.band_names if band not in set(common_bands)]
 
         if one_remaining_bands:
-            all_data.insert(0, one.limit_to_bands(one_remaining_bands).image.data)
+            all_data.insert(0, one.subimage(one_remaining_bands).data)
             # This is not necessary, as new_mask already includes
             # at least all the values of one.image.mask because it comes
             # either from one or from the intersection of one and other
             # new_mask |= one.image.mask[0]
             new_bands = one_remaining_bands + new_bands
 
         if other_remaining_bands:
-            all_data.append(other.limit_to_bands(other_remaining_bands).image.data)
+            all_data.append(other.subimage(other_remaining_bands).data)
             # Apply "or" to the mask in the same way rasterio does, see
             # https://mapbox.github.io/rasterio/topics/masks.html#dataset-masks
             new_mask |= other.image.mask[0]
             new_bands = new_bands + other_remaining_bands
-
         new_image = np.ma.MaskedArray(
             np.concatenate(all_data),
-            mask=new_mask[None].repeat(len(new_bands), axis=0) == 1
+            mask=[new_mask]*len(new_bands)
```

Comment:
are you sure this is what you where looking for, this will keep a reference to `new_mask` the `len(new_bands)` times.
`mask[0] is mask[1]` will return `True`
`new_mask is mask[n]` will be `True` for any n